### DISCUSSION TITLE
stacking array without copying

### DISCUSSION THREAD

**Discussion by daniellga** [rust-ndarray/ndarray#1177] [General]
Hi! Having an array like `[0, 1, 2, 3, 4, 5, 6]`, is it possible to create an array like the one below, without cloning the original (thus I want an ArrayView)?
```
[[0, 2, 4],
 [1, 3, 5],
 [2, 4, 6]]
```

Note that the result is just a combination (stack) of views from the first array, taking a step of 2 between each element.

**Comment by bluss** [comment]
Reading it, it looks like it has column stride of 1 and row stride of 2, which should be possible as an ArrayView and denied for ArrayViewMut - not sure if the ArrayView constructors that take strides allow it, or if they need to be changed to allow this (it's not safe for mutable views since elements overlap.)

**Reply by daniellga** [reply]
You mean like this?
```
fn main() {
    let arr = [0, 1, 2, 3, 4, 5, 6];
    let result = ArrayView::from_shape((3, 3).strides((2, 1)), &arr).unwrap();
}
```
I get the following error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ShapeError/Unsupported: unsupported operation', src/main.rs:10:65
note: [run with `RUST_BACKTRACE=1` environment variable to display a backtrace](https://play.rust-lang.org/#)
```

**Reply by bluss** [reply]
It should be `(3, 3).strides((1, 2))` but it's symmetric so it will fail the same way. Missing feature. It's been discussed in other issues somewhere.

**Reply by jturner314** [reply]
Fwiw, a temporary workaround is to use the unsafe `from_shape_ptr` method. (`from_shape_ptr` has fewer checks than `from_shape`.)

```rust
use ndarray::prelude::*;

fn main() {
    let a = [0, 1, 2, 3, 4, 5, 6];
    let v = unsafe { ArrayView::from_shape_ptr([3, 3].strides([1, 2]), a.as_ptr()) };
    println!("{:?}", v);
}
```

**Reply by daniellga** [reply]
Nice! Thanks! Is it possible to convert the same way if `a` was a `ndarray` (ArrayBase) instead of an `array`?

**Reply by daniellga** [reply]
Oh nvm... It works the same way it seems.
